### DISCUSSION TITLE
"Expected time to win" has become grossly misleading for small farmers; suggested reformulation

### DISCUSSION THREAD

**Discussion by mlin** [Chia-Network/chia-blockchain#2109] [Ideas]
Today, a new farmer who has just filled some old drive they have lying around, probably sees **Expected time to win** of a few months. This invites them to visualize a date a few months from today, when they ought to have won 2 XCH by sitting back and letting their old drive farm in the meantime.

However, that's *grossly* optimistic, because of how much the network will grow during such a timeframe. (The [calculation is based on one's *current* proportion of network space](https://github.com/Chia-Network/chia-blockchain/blob/95d6030876fb19f6836c6c6eeb41273cf7c30d93/chia/cmds/farm_funcs.py#L246-L247).) It's reasonable only if one plans to continually add plots to keep pace with total network growth, which is in no way obvious from the current presentation.

I believe [ChiaCalculator](https://chiacalculator.com/)'s Advanced mode is better modeling the time decay.

Instead of **Expected time to win**, may I suggest presenting **Chance to win in the next week** which would be (double-check me) `1-exp(-proportion*604800/average_block_seconds)`. You can adjust it a little more if you can look up total space 7 days ago and interpolate how your proportion will decay over the next week.

While this doesn't prevent the same kind of misconception, at least it doesn't actively invite it.

**Comment by fullet-tortuga** [comment]
I agree, the wording must be very carefully chosen when playing with money. As it is now is grossly misleading and we can see that with all the complains in many threads.

**Comment by dillon-wright** [comment]
During plotting in recent weeks despite contiinusly plotting the entire time i've actually seen mine go up instead of down because of just how quickly the network is growing. it looks like there is a bandaid below but this is a very solvable problem. 
![image](https://user-images.githubusercontent.com/29571066/116213633-4e3e5c80-a789-11eb-9694-79d86ba1a9fb.png)

**Reply by SackofFluor** [reply]
I think it is save to assume, that the goal to allow "normal" people to do some farming with their spare drives is already dead in the water....

In the last couple of days someone posted that almost 30% or so of all rewards went to one single address.
I have started chia farming around a week ago and in this time the network grew from around 500-600PiB to 1EiB.

So entirely normal people are just not having a realistic change in farming....
yes if you have some TB in spare drives lying around... use them, but for the overwhelming amount of people its just not really feasible.

the GUI is having bugs left and right, Its missing vital information and functions for the "average" Joe.

**Reply by PCDSP** [reply]
0.005/TiB plots / day! LOL! 

This Chia thing is the bee's knee! 
1TB of plots = 15 TiB written on a NVME drive, over many hours... 

So you get to pay some 35€/1TiB of storage to keep that plot, pending you have some 200TiB minimum to earn 1XCH/day - now that is, because tomorrow is another day and 200TiB will give you less then 1 XCH.

LOL! brilliant ponzy scheme, I hope everyone stops farming to have the last laugh, with such 'MASSIVE' coin emission, we're looking at being Bram Cohen's fanclub to keep the mainnet busy!

**Reply by PCDSP** [reply]
> During plotting in recent weeks despite contiinusly plotting the entire time i've actually seen mine go up instead of down because of just how quickly the network is growing. it looks like there is a bandaid below but this is a very solvable problem.
> ![image](https://user-images.githubusercontent.com/29571066/116213633-4e3e5c80-a789-11eb-9694-79d86ba1a9fb.png)

yeah, that little question mark and the amount of growth on the network made me simply say BYE BYE! :)

**Comment by fullet-tortuga** [comment]
You are so right. Sales of hard disks are sky rocketing all over as big whales are stock piling as much as they can. I still wonder how they can mine with all the bugs this program has. It's actually amazing they manage to run it in a large scale already.

**Comment by acalca** [comment]
![projection](https://user-images.githubusercontent.com/4617355/116991978-a6b7b180-acd5-11eb-89df-05e02b243dca.png)
Am I wrong by assuming that the chances of winning are determined by a binomial distribution function?
I wrote a script to calculate the expected time to win considering two netspace growth scenarios. You might find it useful.
I am sorry that it is written in MATLAB but it can be translated to Python in just a few minutes. Just change the extension from .txt to .m to run the code.
[projections.txt](https://github.com/Chia-Network/chia-blockchain/files/6420449/projections.txt)

**Reply by AMSantosP** [reply]
My estimated time to win is 13 years according to the app farming module :-(

**Reply by PCDSP** [reply]
You've done your part, like 95% others...

Now move along and wait until it reaches 1$ / coin :)

**Reply by HeyaGlitz** [reply]
> 
> 
> My estimated time to win is 13 years according to the app farming module :-(

yeah yeah I have 168 years, I'll be dead by then

**Comment by Christoffer2020** [comment]
Is there a website which calculates this winrate/probability as a function of space alloted to your plots? As long as this is only solo, and no pool, maybe it is not feasible for people like me with just one measly plot of 101 GiB.

**Reply by PCDSP** [reply]
keep computer connected for 5 generations, 24/7. you might earn 0.000001 XCH!

**Reply by acalca** [reply]
> keep computer connected for 5 generations, 24/7. you might earn 0.000001 XCH!

If you win, you earn 2XCH. Go to sleep dude...

**Reply by acalca** [reply]
@Christoffer2020 check out the reply to @AMSantosP

**Reply by PCDSP** [reply]
But you never win dude, NEVER. You go into a pool and you earn 0.00000000000000000000000001 or whatever. 

You cannot possibly WIN 1 whole block with just 1 plot or 10 plots for that matter!

**Reply by PCDSP** [reply]
And I'm probably old enough to be your father. Chia is truly designed for clueless millenials! You can use DUDE with your Chia farming buddies! :)

**Comment by AMSantosP** [comment]
https://chiacalculator.com/



Antonio Santos
Sent from my iPhone

On 4 May 2021, at 7:04 PM, Christoffer2020 ***@***.***> wrote:

﻿

Is there a website which calculates this winrate/probability as a function of space alloted to your plots? As long as this is only solo, and no pool, maybe it is not feasible for people like me with just one measly plot of 101 GiB.

**Reply by pluto2021** [reply]
This calculator gives a false image of return without a pool. Even with a pool, the real chance of winning would also depend on the pool size.  Monte Carlo simulator might be better suited for this estimation. In reality, what one really care is the time it takes to have say 90% probability of winning, taking into account all the variables, not a daily winning $ amount, as this is assuming a unity pool of all spaces in the world combined, which is not the case.

**Reply by acalca** [reply]
> website which calculates this winrate/probability as a function of space alloted to your plots? As long as this is only solo, and no pool, maybe it is not feasible for people like me with just one measly plot of 101 GiB.

You can either use the code I uploaded yesterday or you can check the google sheet mentioned (in the comments) here: https://chiadecentral.com/will-i-win-chia-with-2-tb-of-plots/  (I haven't checked it myself though)

You can also calculate it yourself. Say n is your number of plots, then n x 101 GiB is your farm size. Let's say you want to know what are your probabilities (p) of winning one block (2XCH) after d days (there are 4608 chances of winning per day) when the netspace is  2.11EiB (=2.11 x 1024^3 GiB). Your chances of winning are:

p = 1-(1-(n x 101)/(2.11 x 1024^3))^(d x 4608)

So for example, your chances now of winning one block with only 2 plots in 30 days are:

p =  1 - (1-2 x 101/2,2656e+9)^(30 x 4608) = 0.012 = 1.2%

Of course the network will continue to grow, probably exponentially for some time. So, unless you continue plotting, your probability of winning will quickly dissolve in the coming weeks. The code I uploaded yesterday is an attempt to include a projection of how the netspace could grow.

**Reply by pluto2021** [reply]
Probability is one thing, as shown, very simple math. However, lets ask the question differently like this: what is the time it would take to have say 50%, 75%, 90% probability of winning? When you have to do solo farming, this time matters more comparing with a slim probability of winning say in 30 days.  The GUI gives an very rough estimate of the time it would take to win, probably that's only 50%-ish probability. However, statistically, no one has really done the math to allow one to input a fixed probability of winning, and output the time it takes to get there.  This is what's missing in the GUI interface and this calculator.

The chiacalculator is grossly misleading as no solo farmer would be wining everyday for a fraction of a XCH.

**Reply by PCDSP** [reply]
Someone with 50TiB of plots posted a screenshot on May 1st, he'd been farming for three weeks and won 0 XCH!

Thing is there are too many variables, little clarity, add ICO on top and you have all ingredients of a  ponzy scheme.. eco friendly as burning tyres as most miners/farmers are from China and they burn coal for electricity!:)

**Reply by PCDSP** [reply]
I am pleased the value is dropping constantly as this nonsense should never have any impact on SSD prices, that is all I care about.

Thank you Bram for BitTorrent. Pi$$ o55 with Chia nonsense...

**Reply by cr1za** [reply]
> p = 1-(1-(n x 101)/(2.11 x 1024^3))^(d x 4608)
> 
> So for example, your chances now of winning one block with only 2 plots in 30 days are:
> 
> p = 1 - (1-2 x 101/2,2656e+9)^(30 x 4608) = 0.012 = 1.2%

Now how could I roughly calculate from this the daily(or monthly)/XCH I would earn (let's say I'm in a pool resp. winning is distributed evenly)?

**Comment by PCDSP** [comment]
It's a ponzy scheme! Anyone keeping their storage in operation is actually a part of the existence of the coin therefore should receive some form of guaranteed minimum. But the calculator shows a BIG written 'estimated X' and a ? with 'may take up to 3-4 times'. Adding the Netspace growth there, never never earn..

This is gambling/scam and will render your work, time and expense with no return, on the other hand, to advertise:

'green'
'recylce storage space'
'anyone can farm' 
'eco-friendly'

Pay some influencers, to spread the rumour, some jolly millenials with no clue, infect the internet with hope-a-titis, misrepresent the facts and complex implications of running even 100TiB 24/7, not 1PiB!

Get some ad space from showing WD external hdd on chiacalculator.... lol 


THIS IS THE HANDBOOK of a con! 


RIP XCH! 0.50c / coin!

**Reply by SackofFluor** [reply]
@PCDSP 
Ok we got it, you are not convinced that it can work out.
But dude.... really in almost every threat? Do you have anything else to do?

**Comment by PCDSP** [comment]
Lol yes ofc, anything but keeping plots online!

**Comment by LpmRaven** [comment]
> Today, a new farmer who has just filled some old drive they have lying around, probably sees **Expected time to win** of a few months. This invites them to visualize a date a few months from today, when they ought to have won 2 XCH by sitting back and letting their old drive farm in the meantime.
> 
> However, that's _grossly_ optimistic, because of how much the network will grow during such a timeframe. (The [calculation is based on one's _current_ proportion of network space](https://github.com/Chia-Network/chia-blockchain/blob/95d6030876fb19f6836c6c6eeb41273cf7c30d93/chia/cmds/farm_funcs.py#L246-L247).) It's reasonable only if one plans to continually add plots to keep pace with total network growth, which is in no way obvious from the current presentation.
> 
> I believe [ChiaCalculator](https://chiacalculator.com/)'s Advanced mode is better modeling the time decay.
> 
> Instead of **Expected time to win**, may I suggest presenting **Chance to win in the next week** which would be (double-check me) `1-exp(-proportion*604800/average_block_seconds)`. You can adjust it a little more if you can look up total space 7 days ago and interpolate how your proportion will decay over the next week.
> 
> While this doesn't prevent the same kind of misconception, at least it doesn't actively invite it.

@mlin Could you give a working example of your `Chance to win in the next week` calculation, please? I'm not smart enough to understand.
